boa/src/property/mod.rs

```diff
@@ -662,3 +662,10 @@ impl PartialEq<&str> for PropertyKey {
         }
     }
 }
+
+#[derive(Debug, Clone, Copy)]
+pub enum PropertyNameKind {
+    Key,
+    Value,
+    KeyValue,
```

Comment:
```suggestion
    KeyAndValue,
```
Better name for the `Key+Value` variant would be `KeyAndValue` instead of `KeyValue`